src/Octoshift/Services/AwsApi.cs

```diff
@@ -45,7 +55,23 @@ private static AmazonS3Client BuildAmazonS3Client(string awsAccessKeyId, string
 
     public virtual async Task<string> UploadToBucket(string bucketName, string fileName, string keyName)
     {
-        await _transferUtility.UploadAsync(fileName, bucketName, keyName);
+        try
+        {
+            using var cts = new CancellationTokenSource(TimeSpan.FromMinutes(10));
+
+            await _transferUtility.UploadAsync(fileName, bucketName, keyName, cts.Token);
+        }
+        catch (TaskCanceledException ex)
+        {
+            var message = ex.CancellationToken.IsCancellationRequested ? "cancelled" : "timed out";
+
+            throw new OctoshiftCliException($"Upload of archive \"{fileName}\" to AWS {message}", ex);
+        }
+        catch (TimeoutException ex)
```

Comment:
If we hit the 10 (?) min timeout, do we hit a TaskCanceled Exception or a TimeoutException?

Reply:
The 10 min timeout is on the Cancellation Token and would be a TaskCanceledException.

Reply:
So do we need to handle both exceptions if it's always going to be a TaskCanceledException. I'm struggling to understand when it would be one vs the other (also struggling to understand why we need to set the timeout in 2 places, and are using different values for each).

Reply:
When I tested by setting the Timeout on the S3 client to 1 ms, UploadAsync threw a TimeoutException wrapping a TaskCanceledException:

<img width="1023" alt="cancelled" src="https://github.com/github/gh-gei/assets/32278992/7940fbfb-c67f-49ea-a255-cb1b6534be21">

That's why I'm also catching the TimeoutException.